apps/content-publishing-api/src/api.service.ts

```diff
@@ -33,10 +34,36 @@ export class ApiService {
     @InjectRedis() private redis: Redis,
     @InjectQueue(QueueConstants.REQUEST_QUEUE_NAME) private requestQueue: Queue,
     @InjectQueue(QueueConstants.ASSET_QUEUE_NAME) private assetQueue: Queue,
+    @InjectQueue(QueueConstants.PUBLISH_QUEUE_NAME) private publishQueue: Queue,
   ) {
     this.logger = new Logger(this.constructor.name);
   }
 
+  async enqueueContent(
+    schemaId: number,
+    contentDto: OnChainContentDto,
+    onBehalfOf: string,
+  ): Promise<AnnouncementResponseDto> {
+    const jobData: IPublisherJob = {
+      id: Date.now().toString(), // timestamp will get overwritten, but guarantees id-uniqueness
+      schemaId,
+      data: {
+        onBehalfOf,
+        payload: contentDto.payload,
+      },
+    };
+    jobData.id = this.calculateJobId(jobData);
+    const job = await this.publishQueue.add(`OnChain content job - ${jobData.id}`, jobData, {
+      jobId: jobData.id,
+      removeOnComplete: 1000,
+      attempts: 3,
```

Comment:
I don't have strong feelings about this but having attempt in here might be a little be tricky. Since this is in the api we probably want it to fail fast. Because the caller can always retry the call itself

Reply:
This was the existing standard in the app. I'll check the failure conditions, but we do sanity checks before submitting to the chain, so mostly a failure would be for network or Capacity reasons, in which case I think it's reasonable to retry (Capacity exhaustion would pause the queue, anyway).

I'll take a pass through the code & see if I can trap any specific chain errors that we would want to trigger a permanent failure, if we're not already doing that.

Reply:
Okay, so the only _extrinsic_ failure that would cause a retry is `TooManyMessagesInBlock`, which seems reasonable. Any other _extrinsic_ failure explicitly skips the retry.

Other failures, like network errors, etc, _do_ cause a retry, which also seems reasonable.